homeassistant/components/hunterdouglas_powerview/cover.py

```diff
@@ -77,41 +58,34 @@ async def async_setup_entry(
     pv_entry: PowerviewEntryData = hass.data[DOMAIN][entry.entry_id]
     coordinator: PowerviewShadeUpdateCoordinator = pv_entry.coordinator
 
-    entities: list[ShadeEntity] = []
-    for raw_shade in pv_entry.shade_data.values():
-        # The shade may be out of sync with the hub
-        # so we force a refresh when we add it if possible
-        shade: BaseShade = PvShade(raw_shade, pv_entry.api)
-        name_before_refresh = shade.name
-        with suppress(asyncio.TimeoutError):
-            async with async_timeout.timeout(1):
-                await shade.refresh()
+    async def _async_refresh_after_import(self, *_: Any) -> None:
+        """Force position refresh shortly after adding.
 
-        if ATTR_POSITION_DATA not in shade.raw_data:
-            _LOGGER.info(
-                "The %s shade was skipped because it is missing position data",
-                name_before_refresh,
-            )
-            continue
-        coordinator.data.update_shade_positions(shade.raw_data)
-        room_id = shade.raw_data.get(ROOM_ID_IN_SHADE)
-        room_name = pv_entry.room_data.get(room_id, {}).get(ROOM_NAME_UNICODE, "")
+        Legacy shades can become out of sync with hub when moved
+        using physical remotes. This also allows reducing speed
+        of calls to older generation hubs in an effort to
+        prevent hub crashes.
+        """
+
+        for shade in pv_entry.shade_data.values():
+            with suppress(asyncio.TimeoutError):
+                async with async_timeout.timeout(10):
+                    _LOGGER.debug("Initial refresh of shade: %s", shade.name)
+                    await shade.refresh()
+
+    async_call_later(hass, RESYNC_DELAY, _async_refresh_after_import)
```

Comment:
If they reload the config entry before the delay, this will still fire.

Reply:
any hints on how i could resolve that ? happy to review code of other integrations who might be doing the same sort of thing myself

this considerably increased my boot time for HA and I have a small Powerview setup (down to 7seconds from 40+)

Reply:
It returns a callback that when called cancels it.

```
cancel_resync = async_call_later(hass, RESYNC_DELAY, _async_refresh_after_import)

def _async_cancel_resync_if_not_yet_run():
   nonlocal resynced
   if not resynced:  # flip this above when resync is called
      cancel_resync()

entry.async_on_unload(_async_cancel_resync_if_not_yet_run)
```